harness/api_checkers.go

```diff
@@ -0,0 +1,400 @@
+// Package harness provides a suite of API compatibility checks. They were originally
+// ported from the OpenTracing Python library's "harness" module.
+package harness
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"time"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// APICheckCapabilities describes options used by APICheckSuite when testing a Tracer.
+type APICheckCapabilities struct {
```

Comment:
OK I gave a try at implementing functional options based on Tracer.StartSpan — let me know what you think.